packages/sandbox/src/cloudwatch_logs_monitor.ts

```diff
@@ -0,0 +1,297 @@
+   * name for to associate the log group to make it easier
+   * for the user to identify which log groups are being monitored
+   * @param friendlyResourceName The friendly name of the resource that is being monitored
+   * @param logGroupName The log group to read events from
+   */
+  public addLogGroups = (
+    friendlyResourceName: string,
+    logGroupName: string
+  ): void => {
+    this.allLogGroups.push({
+      logGroupName,
+      cloudWatchEventDisplay: {
+        [logGroupName]: {
+          friendlyName: friendlyResourceName,
+          color: this.getNextColorForLogGroup(),
+        },
+      },
+      startTime: this.startTime,
+    });
+  };
+
+  /**
+   * If output location file is specified, the logs will be appended to that file.
+   * If the file doesn't exist it will be created.
+   * @param outputLocation file location
+   */
+  public setOutputLocation = (outputLocation: string) => {
+    if (!outputLocation) {
+      return;
+    }
+    const targetPath = path.isAbsolute(outputLocation)
+      ? outputLocation
+      : path.resolve(process.cwd(), outputLocation);
+    this.printer = new Printer(
+      LogLevel.INFO,
+      fs.createWriteStream(targetPath, { flags: 'a', autoClose: true })
+    );
+  };
+
+  /**
+   * Pick the next color in the object `colors` in round robin fashion
+   */
+  private getNextColorForLogGroup = () => {
+    const colorNames = Object.keys(colors);
+    return colors[
+      colorNames[
+        this.allLogGroups.length % colorNames.length
+      ] as keyof typeof colors
+    ];
+  };
+
+  private scheduleNextTick = (sleep: number): void => {
+    setTimeout(() => void this.tick(), sleep);
+  };
+
+  private tick = async (): Promise<void> => {
+    if (!this.active) {
+      return;
+    }
+    try {
+      const events = (await this.readNewEvents()).flat();
+      events.forEach((event: CloudWatchLogEvent) => {
+        this.print(event);
+      });
+    } catch (e) {
+      throw new AmplifyFault(
+        'FailedToMonitorCloudWatchFault',
+        { message: `Error occurred while monitoring logs: %s` },
+        e instanceof Error ? e : undefined
+      );
+    }
+
+    this.scheduleNextTick(SLEEP);
+  };
+
+  /**
+   * Reads all new log events from a set of CloudWatch Log Groups in parallel
+   */
+  private readNewEvents = async (): Promise<
+    Array<Array<CloudWatchLogEvent>>
+  > => {
+    const promises: Array<Promise<Array<CloudWatchLogEvent>>> = [];
+    for (const cloudWatchLogsToMonitor of this.allLogGroups) {
+      promises.push(this.readEventsFromLogGroup(cloudWatchLogsToMonitor));
+    }
+    return Promise.all(promises);
+  };
+
+  /**
+   * Print out one CloudWatch event using the local printer.
+   */
+  private print = (event: CloudWatchLogEvent): void => {
+    const cloudWatchEventDisplay = this.allLogGroups.find(
+      (logGroup) => logGroup.logGroupName === event.logGroupName
+    )?.cloudWatchEventDisplay;
```

Comment:
We should store this lookup as a map rather than iterating over the array on every print call